Content.Server/_Starlight/GameTicking/Rules/VariationPass/MaintenanceMonstersVariationPassSystem.cs

```diff
@@ -0,0 +1,46 @@
+using System.Linq;
+using Content.Server.GameTicking.Rules.VariationPass.Components;
+using Content.Server.Storage.EntitySystems;
+using Content.Shared.Storage.Components;
+using Content.Shared.EntityTable;
+using Robust.Shared.Map;
+using Robust.Shared.Random;
+
+namespace Content.Server.GameTicking.Rules.VariationPass;
+
+/// <summary>
+/// Handles putting things in lockers around the station, intended for creatures.
+/// </summary>
+public sealed class MaintenanceMonstersVariationPassSystem : VariationPassSystem<MaintenanceMonstersVariationPassComponent>
+{
+    [Dependency] private readonly EntityStorageSystem _entityStorage = default!;
+    [Dependency] private readonly EntityTableSystem _entityTable = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    protected override void ApplyVariation(Entity<MaintenanceMonstersVariationPassComponent> ent, ref StationVariationPassEvent args)
+    {
+        var wiresCut = 0;
+        var query = AllEntityQuery<RoundstartMonsterSpawnComponent, EntityStorageComponent, TransformComponent>();
+        while (query.MoveNext(out var uid, out _, out var storage, out var transform))
+        {
+            // Ignore if not part of the station
+            if (!IsMemberOfStation((uid, transform), ref args))
+                continue;
+
+            // If we don't hit the random chance for this one, skip
+            if (!_random.Prob(ent.Comp.PerLockerProbability))
+                continue;
+
+            var protoNames = _entityTable.GetSpawns(ent.Comp.SpawnTable).Select(x => (string?)(x.ToString())).ToList();
+            var spawns = EntityManager.SpawnEntities(MapCoordinates.Nullspace, protoNames);
+
+            foreach (var spawn in spawns)
+            {
+                if (!_entityStorage.Insert(spawn, uid, storage))
+                {
+                    Del(spawn);
+                }
+            }
```

Comment:
```suggestion
            var protos = _entityTable.GetSpawns(ent.Comp.SpawnTable);
            
            foreach (var proto in protos)
            {
                var spawn = Spawn(proto, MapCoordinates.Nullspace);

                if (!_entityStorage.Insert(spawn, uid, storage))
                {
                    Del(spawn);
                }
            }
```


`GetSpawns`  already returns a list of prototype IDs (EntProtoId`s) so we can really simplify this
SpawnEntities is a spawn loop through list of prototypes, here we do the same loop but directly, without the excessive LINQ(**evil**) 

> (If you would appply this patch, remove LINQ using aswell)

```C#
    public EntityUid[] SpawnEntities(MapCoordinates coordinates, params string?[] protoNames)
    {
        var ents = new EntityUid[protoNames.Length];
        for (var i = 0; i < protoNames.Length; i++)
        {
            ents[i] = Spawn(protoNames[i], coordinates);
        }
        return ents;
    }
```

Reply:
I've changed it up to do so, and added in an early break if we start failing to add an entity to the lockers.